src/transformers/modeling_squeezebert.py

```diff
@@ -0,0 +1,1037 @@
+        # seem a bit unusual, but is taken from the original Transformer paper.
+        attention_probs = self.dropout(attention_probs)
+
+        context_layer = self.matmul_qkv(attention_probs, value_layer)
+        context_layer = self.transpose_output(context_layer)
+
+        result = {"context_layer": context_layer}
+        if output_attentions:
+            result["attention_score"] = attention_score
+        return result
+
+
+class SqueezeBertModule(nn.Module):
+    def __init__(self, config):
+        """
+        hidden_size = input chans = output chans for Q, K, V (they are all the same ... for now) = output chans for the module
+        intermediate_size = output chans for intermediate layer
+        groups = number of groups for all layers in the BertModule. (eventually we could change the interface to allow
+                 different groups for different layers)
+        """
+        super().__init__()
+
+        c0 = config.hidden_size
+        c1 = config.hidden_size
+        c2 = config.intermediate_size
+        c3 = config.hidden_size
+
+        self.attention = SqueezeBertSelfAttention(
+            config=config, cin=c0, q_groups=config.q_groups, k_groups=config.k_groups, v_groups=config.v_groups
+        )
+        self.post_attention = CDL(
+            cin=c0, cout=c1, groups=config.post_attention_groups, dropout_prob=config.hidden_dropout_prob
+        )
+        self.intermediate = CA(cin=c1, cout=c2, groups=config.intermediate_groups, act=config.hidden_act)
+        self.output = CDL(cin=c2, cout=c3, groups=config.output_groups, dropout_prob=config.hidden_dropout_prob)
+
+    def forward(self, hidden_states, attention_mask, output_attentions):
+        att = self.attention(hidden_states, attention_mask, output_attentions)
+        attention_output = att["context_layer"]
+
+        post_attention_output = self.post_attention(attention_output, hidden_states)
+        intermediate_output = self.intermediate(post_attention_output)
+        layer_output = self.output(intermediate_output, post_attention_output)
+
+        result = {"feature_map": layer_output}
+        if output_attentions:
+            result["attention_score"] = att["attention_score"]
+
+        return result
+
+
+class SqueezeBertEncoder(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        assert config.embedding_size == config.hidden_size, (
+            "If you want embedding_size != intermediate hidden_size,"
+            "please insert a Conv1d layer to adjust the number of channels "
+            "before the first SqueezeBertModule."
+        )
+
+        layers = [SqueezeBertModule(config) for _ in range(config.num_hidden_layers)]
+        self.layers = nn.ModuleList(*[layers])
+
+    def forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        head_mask=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
```

Comment:
You don't need to put these in here if they're not implemented

Reply:
I was able to remove `encoder_hidden_states` and `encoder_attention_mask` from the input arguments of SqueezeBertModel.

Unfortunately, the tests in `test_modeling_common.py` do input `head_mask=None` even when `test_head_masking==False`. I haven't determined what line of code causes this to happen. So, I am leaving `head_mask` in for now, and I assert that `head_mask is None`.

Reply:
Indeed, we should update this. Thanks!